packages/servicenow_cmdb/data_stream/event/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,3243 @@
+      message: error message set and no data to process.
+  - rename:
+      field: message
+      tag: rename_message_to_event_original
+      target_field: event.original
+      ignore_missing: true
+      description: Renames the original `message` field to `event.original` to store a copy of the original message. The `event.original` field is not touched if the document already has one; it may happen when Logstash sends the document.
+      if: ctx.event?.original == null
+  - remove:
+      field: message
+      tag: remove_message
+      ignore_missing: true
+      description: The `message` field is no longer required if the document has an `event.original` field.
+      if: ctx.event?.original != null
+  - json:
+      field: event.original
+      tag: json_event_original
+      target_field: servicenow_cmdb.event
+      if: ctx.event?.original != null
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.on_failure_pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
+  - rename:
+      field: _conf.table_name
+      tag: rename_table_name
+      target_field: servicenow_cmdb.event.table_name
+      ignore_missing: true
+      if: ctx.servicenow_cmdb?.event?.table_name == null
+  - fingerprint:
+      fields:
+        - servicenow_cmdb.event.sys_id.value
+        - servicenow_cmdb.event.sys_updated_on.value
+        - servicenow_cmdb.event.table_name
+      target_field: _id
+      ignore_missing: true
+  - script:
+      tag: script_to_drop_undefined_values
+      lang: painless
+      description: Drops undefined values recursively.
+      source: |-
+        boolean drop(Object object) {
+          if ((object instanceof String && ((String) object).equalsIgnoreCase('unknown')) || (object instanceof String && ((String) object).equalsIgnoreCase('none')) || (object instanceof String && ((String) object).equalsIgnoreCase('null')) || (object instanceof String && ((String) object).equalsIgnoreCase('n/a')) || (object instanceof String && ((String) object).equalsIgnoreCase('na'))) {
+            return true;
+          } else if (object instanceof Map) {
+            ((Map) object).values().removeIf(v -> drop(v));
+            return (((Map) object).size() == 0);
+          } else if (object instanceof List) {
+            ((List) object).removeIf(v -> drop(v));
+            return (((List) object).length == 0);
+          }
+          return false;
+        }
+        drop(ctx);
+  - set:
+      field: event.kind
+      tag: set_event_kind_to_event
+      value: event
+      if: ctx.servicenow_cmdb?.event?.table_name != null && ['cmdb_ci_service','change_task','cmdb_rel_ci','cmdb_ci_infra_service','cmdb_ci','cmdb','cmdb_ci_vm','cmdb_ci_db_instance','cmdb_ci_appl','cmdb_ci_esx_server'].contains(ctx.servicenow_cmdb.event.table_name)
```

These allocate each time they are executed. They can be expressed as a set of disjunctions in a condition.
```suggestion
      if: >-
        ctx.servicenow_cmdb?.event?.table_name != null && (
          ctx.servicenow_cmdb.event.table_name == 'cmdb_ci_service' ||
          ctx.servicenow_cmdb.event.table_name == 'change_task' ||
          ctx.servicenow_cmdb.event.table_name == 'cmdb_rel_ci' ||
          ctx.servicenow_cmdb.event.table_name == 'cmdb_ci_infra_service' ||
          ctx.servicenow_cmdb.event.table_name == 'cmdb_ci' ||
          ctx.servicenow_cmdb.event.table_name == 'cmdb' ||
          ctx.servicenow_cmdb.event.table_name == 'cmdb_ci_vm' ||
          ctx.servicenow_cmdb.event.table_name == 'cmdb_ci_db_instance' ||
          ctx.servicenow_cmdb.event.table_name == 'cmdb_ci_appl' ||
          ctx.servicenow_cmdb.event.table_name == 'cmdb_ci_esx_server'
        )
```